include/unifex/manual_lifetime.hpp

```diff
@@ -40,7 +40,7 @@ class manual_lifetime {
   }
 
   template <typename Func>
-  T& construct_from(Func&& func) noexcept(noexcept(T(((Func &&) func)()))) {
```

Comment:
This change is necessary because in the previous formulation MSVC tried to test the `noexcept`-ness of the move constructor, and for operation states, the move constructor is deleted. Since we rely on RVO, the move constructor is never invoked, so we only need to test the `noexcept`-ness of the function call.